daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -733,6 +740,57 @@ private[lf] final class Compiler(
       translateFetchTemplateBody(env, tmplId, tmpl)(cidPos, None, tokenPos)
     }
 
+  @nowarn("msg=parameter value tokenPos.* is never used")
+  private[this] def translateSoftFetchTemplateBody(
+      env: Env,
+      tmplId: Identifier,
+      tmpl: Template,
+      predPids: List[PackageId],
+  )(
+      cidPos: Position,
+      tokenPos: Position,
+  ): s.SExpr =
+    let(
+      env,
+      SBFetchAny(
+        env.toSEVar(cidPos),
+        s.SEValue.None,
+      ),
+    ) { (anyTmplArgPos, env) =>
+      let(
+        env,
+        SBPromoteAnyContract(
+          tmplId,
+          predPids.map(Identifier(_, tmplId.qualifiedName)),
+        )(
+          env.toSEVar(cidPos),
+          env.toSEVar(anyTmplArgPos),
+        ),
+      ) { (tmplArgPos, _env) =>
+        val env = _env.bindExprVar(tmpl.param, tmplArgPos)
+        let(
+          env,
+          SBUInsertFetchNode(tmplId, byKey = false)(env.toSEVar(cidPos)),
```

Comment:
we do it for the side effects 🥲

Reply:
Ohhh! I didn't see the second application! I'm guessing we put it in a let as a way to force it to perform the insert before evaluating the body, because speedy is strict?

Reply:
yes I think so